citest/base/base_test_case.py

```diff
@@ -123,41 +126,63 @@ def __end_step_context(self, relation):
       return
     JournalLogger.end_context(relation=relation)
 
-  def __call__(self, result=None):
+  def __trap_skip(self, delegate, test, reason):
+    self.__final_outcome_relation = 'VALID'
+    return delegate(test, reason)
+
+  def __trap_success(self, delegate, test):
+    self.__final_outcome_relation = 'VALID'
+    return delegate(test)
+
+  def __trap_failure(self, delegate, test, err):
+    self.__final_outcome_relation = 'INVALID'
+    return delegate(test, err)
+
+  def __trap_error(self, delegate, test, err):
+    self.__final_outcome_relation = 'ERROR'
+    error_details = '%s: %s' % (err[0], err[1])
+    JournalLogger.journal_or_log_detail(
+        'Raised Exception', error_details,
+        levelno=logging.ERROR, format='pre', _logger=self.logger)
+    return delegate(test, err)
+
+  def __call__(self, *args, **kwargs):
     """Wraps the base class fixture heuristics that run an individual test."""
+    # It appears this is not standard.
+    # The default test passes args[0]
+    # but py.test passes kwargs 'result'
+    if args:
+      result = args[0]
+    else:
+      result = kwargs.get('result')
+
     if result is None:
       result = self.defaultTestResult()
 
-    num_failures = len(result.failures)
-    num_errors = len(result.errors)
+    do_skip = result.addSkip
+    do_success = result.addSuccess
+    do_error = result.addError
+    do_failure = result.addError
```

Comment:
Do we want `addError` or `addFailure` in this case?

Reply:
Good catch.